File: src/main/java/mel/wadzapi/interview/service/StatTimerService.java
package mel.wadzapi.interview.service;

import mel.wadzapi.interview.controller.dto.EventHappensDto;
import mel.wadzapi.interview.controller.dto.HamStatResponseDto;

import java.time.Instant;

//TODO: javadocs && layering
public interface StatTimerService {
    // Возвращает Instant на начало текущих суток
    Instant today();
    // Обработка входящего события
    void processEvent(EventHappensDto eventHappensDto);
    // Выдача статистики по идентификаторам
    HamStatResponseDto calcHamStat(int... hamsterIds);

}

File: src/main/kotlin/mel/wadzapi/interview/service/StatTimerService.kt
package mel.wadzapi.interview.service

import mel.wadzapi.interview.controller.dto.EventHappensDto
import mel.wadzapi.interview.controller.dto.HamStatResponseDto
import java.time.Instant

//TODO: javadocs && layering
interface StatTimerService {
    // Возвращает Instant на начало текущих суток
    fun today(): Instant?

    // Обработка входящего события
    fun processEvent(eventHappensDto: EventHappensDto)

    // Выдача статистики по идентификаторам
    fun calcHamStat(vararg hamsterIds: Int): HamStatResponseDto
}